File: LeetCode-December-Challenge/ImplementQueueusingStacks.java
import java.util.*;

class MyQueue {
    private Stack<Integer> s1, s2;

    public MyQueue() {
        s1 = new Stack<>();
        s2 = new Stack<>();
    }

    public void push(int x) {
        s1.push(x);
    }

    public int pop() {
        peek();
        int top =  s2.peek();
        s2.pop();
        return top;
    }

    public int peek() {
        if (s2.empty())
            while (!s1.empty()) {
                s2.add(s1.peek());
                s1.pop();
            }
        return s2.peek();
    }

    public boolean empty() {
        return s1.empty() && s2.empty();
    }
}

public class ImplementQueueusingStacks {
    public static void main(String[] args) {

    }
}

File: Year-2022/LeetCode-December-Challenge/ImplementQueueusingStacks.cpp
#include<bits/stdc++.h>
using namespace std;
class MyQueue {
private:
    stack<int> s1, s2;
public:
    void push(int x) {
        s1.push(x);
    }
    
    int pop() {
        peek();
        int top = s2.top();
        s2.pop();
        return top;
    }
    
    int peek() {
        if(s2.empty())
            while(!s1.empty()) s2.push(s1.top()), s1.pop();
        return s2.top();
    }
    
    bool empty() {
        return s1.empty() && s2.empty();
    }
};
int main(){
    MyQueue *q = new MyQueue();
    q->push(1); 
    q->push(2); 
    cout<<q->peek()<<endl;
    cout<<q->pop()<<endl;
    cout<<std::boolalpha<<q->empty()<<endl;
    return 0;
}